features/messaging/messages-screen.tsx

```diff
@@ -0,0 +1,322 @@
+  const [highlightedMessageId, setHighlightedMessageId] = useState<string>('')
+  const [showInModal, setShowInModal] = useState<'' | 'contextMenu' | 'reactions'>('')
+  const [replying, setReplying] = useState<boolean>(false)
+  const [attachmentOpen, setAttachmentOpen] = useState<boolean>(false)
+  const [imageUrl, setImageUrl] = useState<string>('')
+
+  const conversation = conversations[conversationId]
+  const members = memberships[conversationId].filter((m) => m.expand?.user.id !== user?.id)
+  const ownMembership = memberships[conversationId].filter((m) => m.expand?.user.id === user?.id)[0]
+  const router = useRouter()
+
+  const conversationMessages = messagesPerConversation[conversationId]
+  const highlightedMessage = conversationMessages.find((m) => m.id === highlightedMessageId)
+
+  const colorMap = useMemo(() => {
+    const colors = randomColors(members.length)
+
+    const map = members.reduce((acc, member, index) => {
+      if (member.expand?.user.id) acc[member.expand?.user.id] = colors[index]
+      return acc
+    }, {} as Record<string, any>)
+
+    return map
+  }, [members.length])
+
+  useEffect(() => {
+    const showSub = Keyboard.addListener('keyboardDidShow', () => {
+      //flatListRef.current?.scrollToIndex({ animated: true, index: 0 });
+    })
+    return () => showSub.remove()
+  }, [])
+
+  useEffect(() => {
+    async function setLastRead() {
+      const lastReadDate = conversationMessages[0].created
+      await pocketbase
+        .collection('memberships')
+        .update(ownMembership.id, { last_read: lastReadDate })
+    }
+    setLastRead()
+  }, [])
+
+  const onMessageLongPress = (id: string) => {
+    setHighlightedMessageId(id)
+    setShowInModal('contextMenu')
+  }
+
+  if (!user) return null
+
+  const onMessageSubmit = () => {
+    sendMessage(
+      user.id,
+      conversationId,
+      message,
+      replying ? highlightedMessageId : undefined,
+      imageUrl || undefined
+    )
+    setMessage('')
+    setHighlightedMessageId('')
+    setReplying(false)
+    setAttachmentOpen(false)
+    setImageUrl('')
+    flatListRef.current?.scrollToIndex({ index: 0, animated: true })
+  }
+
+  const onAttachmentPress = () => {
+    setAttachmentOpen(true)
+  }
+
+  const loadMoreMessages = async () => {
+    if (!user) return
+    if (oldestLoadedMessageDate[conversationId] === firstMessageDate[conversationId]) return
+
+    const newMessages = await pocketbase.collection('messages').getList<Message>(0, PAGE_SIZE, {
+      filter: `conversation = "${conversationId}" && created < "${oldestLoadedMessageDate[conversationId]}"`,
+      sort: '-created',
+    })
+    const oldestMessage = newMessages.items[newMessages.items.length - 1]
+    setOldestLoadedMessageDate(conversationId, oldestMessage.created!)
+    addOlderMessages(conversationId, newMessages.items)
+    console.log(
+      'new messages (old)',
+      newMessages.items.map((m) => m.text)
+    )
```

Comment:
We should delete these, there are a few more left

Reply:
It's fine to keep the ones that are recording things like subscriptions, but don't have too many